nutrition_labels/grant_data_processing.py

```diff
@@ -23,216 +24,321 @@ def merge_tags(data, person_cols_list):
-        how='all').reset_index(drop=True)
+        subset=[
+            "Normalised code - RF",
+            "Normalised code - grants",
+            "Normalised code - EPMC",
+        ],
+        how="all",
+    ).reset_index(drop=True)
     # Get the final single code for each grant:
     # If there is conflict in tags, the order of truth is:
     # 1. EPMC (this shows solid evidence of the output)
     # 2. RF (they might not have created the output at the time of the report)
     # 3. Grants (they might not have created the output at the time of the application)
     code = []
     for _, row in grant_data.iterrows():
-        if pd.notnull(row['Normalised code - EPMC']):
-            code.append(row['Normalised code - EPMC'])
-        elif pd.notnull(row['Normalised code - RF']):
-            code.append(row['Normalised code - RF'])
-        elif pd.notnull(row['Normalised code - grants']):
-            code.append(row['Normalised code - grants'])
+        if pd.notnull(row["Normalised code - EPMC"]):
+            code.append(row["Normalised code - EPMC"])
+        elif pd.notnull(row["Normalised code - RF"]):
+            code.append(row["Normalised code - RF"])
+        elif pd.notnull(row["Normalised code - grants"]):
+            code.append(row["Normalised code - grants"])
         else:
-            print('Should have deduplicated this row?')
+            print("Should have deduplicated this row?")
 
     # Map the final codes onto whether the grant is 'relevant' (1) or not (0)
-    relevance_dict = {1:1, 2:1, 3:1, 5:0}
-    grant_data['Relevance code'] = [relevance_dict[int(c)] for c in code]
-
-    # It seems like some descriptions have some spaces deleted, which means
-    # they are classed as different but should really be the same
-    grant_data['Description to deduplicate with'] = grant_data['Description'].str.replace(r'\W','')
-    grant_data = grant_data.drop_duplicates(['Internal ID 6 digit', 'Description to deduplicate with'])
-
-    num_irrelevant = len([i for i in grant_data['Relevance code'] if i==0])
-    print(num_irrelevant)
-    print(len(grant_data) - num_irrelevant)
-    grant_data = grant_data[[
-        'Internal ID', 'RF question', 'RF Name', 'pmid', 'Relevance code',
-        'Normalised code - RF', 'Normalised code - grants', 'Normalised code - EPMC',
-        'Description', 'Title', 'Grant Programme:Title']]
+    relevance_dict = {1: 1, 2: 1, 3: 1, 5: 0}
+    grant_data["Relevance code"] = [relevance_dict[int(c)] for c in code]
+
+    return grant_data
+
+
+def deduplicate_similar_grants(
+    grant_data, text_column="Description", grant_id_column="Internal ID 6 digit"
+):
+    """
+    Some descriptions have some spaces deleted. We don't want to include
+    grants from the same family (6 digit ID) if they have effectively the same description
+    """
+    grant_data = grant_data.copy()
+    grant_data["Temporary column"] = grant_data[text_column].str.replace(r"\W", "")
```

Comment:
is the use of TemporaryColumn necessary?

can you not just do
```
grant_data = grant_data.drop_duplicates([grant_id_column, text_column])
```

is that line necessary? `grant_data["Temporary column"] = grant_data[text_column].str.replace(r"\W", "")` it should not affect the duplication removal